x-pack/plugins/cases/server/client/utils.ts

```diff
@@ -500,7 +477,7 @@ export const getCaseToUpdate = (
         if (!deepEqual(currentValue, value)) {
           acc[key] = value;
         }
-      } else if (currentValue != null && value !== currentValue) {
+      } else if (currentValue !== undefined && value !== currentValue) {
```

Comment:
Strangely enough, we never had to patch a field that could be `null`.

Reply:
Yeah `category` is null by default. The patch tests were failing because the requests were not producing any changes. That was caused by this line, if the SO original value was `null` the new value was not set.